resources/assets/js/components/WordCounter/WordCounter.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from "react";
+import ProgressRing from "../ProgressRing";
+
+/*
+  Question: Should this component have the option to add a textarea component (or input element)?
+  example:
+    prop: textAreaComponent: React.ReactElement
+    ...
+    return (
+      {textAreaComponent && textAreaComponent}
+    )
+*/
+
+interface Message {
+  max: number;
+  message: string;
+}
+
+interface WordCounterProps {
+  numOfWords: number;
+  maxWords: number;
```

Comment:
I'd add some comments for the interface fields. For example, the difference between maxWords, numOfWords, and wordLimit is not obvious. Also add a comment about the message array needing to be ordered from lowest max to largest.

Reply:
Although perhaps this component could itself reorder messages by max, just to be sure.